scripts/tests/yaml/runner.py

```diff
@@ -0,0 +1,294 @@
+    runner_config = TestRunnerConfig(hooks=TestRunnerLogger())
+
+    runner = TestRunner()
+    return runner.run(parser_builder_config, runner_config)
+
+
+@runner_base.command()
+@test_runner_options
+@pass_parser_builder_config
+def run(parser_builder_config: TestParserBuilderConfig, adapter: str, stop_on_error: bool, stop_on_warning: bool, stop_at_number: int, show_adapter_logs: bool, show_adapter_logs_on_error: bool):
+    """Run the test suite."""
+    adapter = __import__(adapter, fromlist=[None]).Adapter(parser_builder_config.parser_config.definitions)
+    runner_options = TestRunnerOptions(stop_on_error, stop_on_warning, stop_at_number)
+    runner_hooks = TestRunnerLogger(show_adapter_logs, show_adapter_logs_on_error)
+    runner_config = TestRunnerConfig(adapter, get_default_pseudo_clusters(), runner_options, runner_hooks)
+
+    runner = TestRunner()
+    return runner.run(parser_builder_config, runner_config)
+
+
+@runner_base.command()
+@test_runner_options
+@websocket_runner_options
+@pass_parser_builder_config
+def websocket(parser_builder_config: TestParserBuilderConfig, adapter: str, stop_on_error: bool, stop_on_warning: bool, stop_at_number: int, show_adapter_logs: bool, show_adapter_logs_on_error: bool, server_address: str, server_port: int, server_path: str, server_name: str, server_arguments: str):
+    """Run the test suite using websockets."""
+    adapter = __import__(adapter, fromlist=[None]).Adapter(parser_builder_config.parser_config.definitions)
+    runner_options = TestRunnerOptions(stop_on_error, stop_on_warning, stop_at_number)
+    runner_hooks = TestRunnerLogger(show_adapter_logs, show_adapter_logs_on_error)
+    runner_config = TestRunnerConfig(adapter, get_default_pseudo_clusters(), runner_options, runner_hooks)
+
+    if server_path is None and server_name:
+        paths_finder = PathsFinder()
+        server_path = paths_finder.get(server_name)
+
+    websocket_runner_hooks = WebSocketRunnerLogger()
+    websocket_runner_config = WebSocketRunnerConfig(
+        server_address, server_port, server_path, server_arguments, websocket_runner_hooks)
+
+    runner = WebSocketRunner(websocket_runner_config)
+    return runner.run(parser_builder_config, runner_config)
+
+
+@runner_base.command()
+@test_runner_options
+@websocket_runner_options
+@click.pass_context
+def chiptool(ctx, *args, **kwargs):
+    """Run the test suite using chip-tool."""
+    return ctx.forward(websocket)
+
+
+@runner_base.command()
+@test_runner_options
+@websocket_runner_options
+@click.pass_context
+def app1(ctx, *args, **kwargs):
+    """Run the test suite using app1."""
+    return ctx.forward(websocket)
+
+
+@runner_base.command()
+@test_runner_options
+@websocket_runner_options
+@click.pass_context
+def app2(ctx, *args, **kwargs):
+    """Run the test suite using app2."""
+    return ctx.forward(websocket)
+
+
+if __name__ == '__main__':
+    success = False
+    try:
+        success = runner_base(standalone_mode=False)
```

Comment:
What is `standalone_mode`, is don't see that anywhere

Reply:
I have added a comment to make it clearer what it is and why it is set to `False`